packages/medusa/src/api/routes/admin/products/transaction/create-product-variant.ts

```diff
@@ -96,7 +94,7 @@ export const createVariantTransaction = async (
       return
     }
 
-    return await inventoryServiceTx!.createInventoryItem({
+    return await inventoryService!.createInventoryItem({
```

Comment:
suggestion: if we put `!` on the service i means that we are sure it will be present. In that case I would remove the `?` on the type

Reply:
Inventory service is indeed optional, and the functions using it are only called if it is present.
`const strategy = inventoryService ? createVariantStrategyWithInventory : createSimpleVariantStrategy`
So this is the expected behavior, but TS doesn't recognize that these functions are called when the variable has value.